File: systems-for-design-and-implementation/labs/Java_ChatSockets/ChatModel/src/chat/model/Message.java
package chat.model;

/**
 * Created by IntelliJ IDEA.
 * User: grigo
 * Date: Mar 18, 2009
 * Time: 1:07:39 PM
 */
public class Message
{
    private User sender, receiver;
    private String text;

    public Message(User sender, String text, User receiver)
    {
        this.sender = sender;
        this.text = text;
        this.receiver = receiver;
    }

    public User getSender()
    {
        return sender;
    }

    public User getReceiver()
    {
        return receiver;
    }

    public String getText()
    {
        return text;
    }

    @Override
    public String toString()
    {
        return "Message{" +
            "sender=" + sender.getId() +
            ", receiver=" + receiver.getId() +
            ", text='" + text + '\'' +
            '}';
    }
}

File: systems-for-design-and-implementation/labs/CSharp_ChatSockets/model/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace chat.model
{   
    public class Message
    {
    private User sender, receiver;
    private String message;

    public Message(User sender, User receiver, string message)
    {
        this.sender = sender;
        this.receiver = receiver;
        this.message = message;
    }

    public User Sender
    {
        get { return sender; }
    }

    public User Receiver
    {
        get { return receiver; }
    }

    public string Text
    {
        get { return message; }
    }

    public override string ToString()
    {
        return string.Format("Sender: {0}, Receiver: {1}, Message: {2}", sender, receiver, message);
    }
    }
}
